### PR TITLE
Replace MultiTableMetadata with Metadata

### LINKED ISSUE #2129: Enable multi table synthesizers to use new Metadata
### Problem Description

_As a user, I'd like to be able to use my metadata wherever it is applicable once created without having to do extra work._

### Expected behavior

Once #2104 is complete, we need to enable multi table synthesizers to work with the new Metadata
- Support the new class on initialization
- Deprecate MultiTableMetadata
    - Raise a future warning if users use this
- They should be backwards compatible (still support the old class)

### Additional context

Since the new class will be almost identical to the current MultiTableMetadata class, this should be a straightforward fix.

### PR DESCRIPTION
**Pull request by lajohn4747**
resolves #2129 
CU-86b19awxk

Replace `MultiTableMetadata` with `Metadata` and warn the user of deprecation. Since `Metadata` is a derived class of `MultiTableMetadata`, almost all instances can be swapped easily. Tests have to change because new instances are created for `Metadata` conversion and it contains the spec version `V1` instead of `MULTI_TABLE_V1`

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b19awxk SDV - Enable multi table synthesizers to use new Metadata #2129](https://app.clickup.com/t/86b19awxk)

File: tests/integration/metadata/test_metadata.py [hunk 1]
```diff
@@ -255,3 +258,41 @@ def test_single_table_compatibility(tmp_path):
     metadata_sample = synthesizer.sample(10)
     assert loaded_synthesizer.metadata.to_dict() == synthesizer_2.metadata.to_dict()
     assert metadata_sample.columns.to_list() == loaded_sample.columns.to_list()
+
+
+def test_multi_table_compatibility(tmp_path):
+    """Test if SingleMetadataTable still has compatibility with single table synthesizers."""
+    # Setup
+    data, metadata = download_demo('multi_table', 'fake_hotels')
```
[the remark this answers is not in the document]
Comment (lajohn4747) [reply tests/integration/metadata/test_metadata.py]: Yep updated to recreate the legacy objects `SingleMetadataTable` and `MultiMetadataTable` from dictionaries so it won't be affected once `download_demo` has been changed

File: tests/integration/metadata/test_metadata.py [hunk 2]
```diff
@@ -274,3 +277,82 @@ def test_single_table_compatibility(tmp_path):
...
+        'METADATA_SPEC_VERSION': 'MULTI_TABLE_V1',
+    }
+    metadata = MultiTableMetadata.load_from_dict(multi_dict)
+    assert type(metadata) is MultiTableMetadata
+
+    # Run
+    with pytest.warns(FutureWarning, match=warn_msg):
+        synthesizer = HMASynthesizer(metadata)
+    synthesizer.fit(data)
+    model_path = tmp_path / 'synthesizer.pkl'
```
Comment (gsheni) [inline tests/integration/metadata/test_metadata.py:336]: Use os.path.join

**Review by gsheni — APPROVED** [review]
lgtm

File: tests/integration/metadata/test_metadata.py (hunk 2 above)
Comment (gsheni) [reply tests/integration/metadata/test_metadata.py]: Fixed:
https://github.com/sdv-dev/SDV/pull/2146/commits/b88355ed44ac38c7f5f5f7c0ffc3a3a83b57d517

File: tests/integration/metadata/test_metadata.py [hunk 3]
```diff
@@ -274,3 +278,82 @@ def test_single_table_compatibility(tmp_path):
...
+            }
+        ],
+        'METADATA_SPEC_VERSION': 'MULTI_TABLE_V1',
+    }
+    metadata = MultiTableMetadata.load_from_dict(multi_dict)
+    assert type(metadata) is MultiTableMetadata
+
+    # Run
+    with pytest.warns(FutureWarning, match=warn_msg):
+        synthesizer = HMASynthesizer(metadata)
```
Comment (pvk-developer) [inline tests/integration/metadata/test_metadata.py:335]: add breakline.

File: tests/integration/metadata/test_metadata.py [hunk 4]
```diff
@@ -274,3 +278,82 @@ def test_single_table_compatibility(tmp_path):
...
+    assert os.path.isfile(model_path)
+    loaded_synthesizer = HMASynthesizer.load(model_path)
+    assert isinstance(synthesizer, HMASynthesizer)
+    assert loaded_synthesizer.get_info() == synthesizer.get_info()
+    assert isinstance(loaded_synthesizer.metadata, Metadata)
+    expected_metadata = metadata.to_dict()
+    expected_metadata['METADATA_SPEC_VERSION'] = 'V1'
+    assert loaded_synthesizer.metadata.to_dict() == expected_metadata
+    loaded_sample = loaded_synthesizer.sample(10)
+    synthesizer.validate(loaded_sample)
```
Comment (pvk-developer) [inline tests/integration/metadata/test_metadata.py:351]: Could we split this into multiple sections:
```
# Load HMASynthesizer
# Asserts
# Sample from loaded synthesizer
```

File: sdv/multi_table/base.py [hunk 5]
```diff
@@ -38,8 +40,8 @@ class BaseMultiTableSynthesizer:
     multi table synthesizers need to implement, as well as common functionality.
 
     Args:
-        metadata (sdv.metadata.multi_table.MultiTableMetadata):
-            Multi table metadata representing the data tables that this synthesizer will be used
+        metadata (sdv.metadata.Metadata):
+            Table metadata representing the data tables that this synthesizer will be used
```
Comment (frances-h) [inline sdv/multi_table/base.py:44]: Maybe rephrase to just "Metadata representing the..."?

File: tests/integration/metadata/test_metadata.py [hunk 6]
```diff
@@ -274,3 +278,82 @@ def test_single_table_compatibility(tmp_path):
     metadata_sample = synthesizer.sample(10)
     assert loaded_synthesizer.metadata.to_dict() == synthesizer_2.metadata.to_dict()
     assert metadata_sample.columns.to_list() == loaded_sample.columns.to_list()
+
+
+def test_multi_table_compatibility(tmp_path):
+    """Test if SingleMetadataTable still has compatibility with single table synthesizers."""
```
Comment (frances-h) [inline tests/integration/metadata/test_metadata.py:284]: Typo: MultiTableMetadata, multi table synthesizers

Comment (gsheni) [reply tests/integration/metadata/test_metadata.py]: Fixed

File: sdv/multi_table/base.py (hunk 5 above)
Comment (gsheni) [reply sdv/multi_table/base.py]: Fixed

File: tests/integration/metadata/test_metadata.py (hunk 4 above)
Comment (gsheni) [reply tests/integration/metadata/test_metadata.py]: Fixed

File: tests/integration/metadata/test_metadata.py (hunk 3 above)
Comment (gsheni) [reply tests/integration/metadata/test_metadata.py:335]: Fixed

**Review by amontanez24 — APPROVED** [review]

**Review by frances-h — APPROVED** [review]
